src/main/webapp/app/exam/manage/exams/exam-update.component.ts

```diff
@@ -44,99 +43,163 @@ export class ExamUpdateComponent implements OnInit {
+     * Returns to the overview page if there is no previous state, and we created a new exam
      */
-    previousState() {
+    resetToPreviousState() {
         this.navigationUtilService.navigateBackWithOptional(['course-management', this.course.id!.toString(), 'exams'], this.exam.id?.toString());
     }
 
+    /**
+     * Calculates the WorkingTime for real exams based on the start- and end-time.
+     */
+    calculateWorkingTime() {
+        if (this.exam.testExam) return;
+
+        if (this.exam.startDate && this.exam.endDate) {
+            this.exam.workingTime = dayjs(this.exam.endDate).diff(this.exam.startDate, 's');
+        } else {
+            this.exam.workingTime = 0;
+        }
+    }
+
+    /**
+     * Returns the maximum working time in minutes for test exams.
+     */
+    get maxWorkingTimeInMinutes(): number {
+        if (!this.exam.testExam) return 0;
+
+        if (this.exam.startDate && this.exam.endDate) {
+            return dayjs(this.exam.endDate).diff(this.exam.startDate, 'm');
+        } else {
+            // In case of an import, the exam.workingTime is imported, but the start / end date are deleted -> no error should be shown to the user in this case
+            return this.isImport ? this.workingTimeInMinutes : 0;
+        }
+    }
+
+    /**
+     * Saves the exam. If the dates have changed and the exam is ongoing, a confirmation modal is shown to the user.
+     * If either the user confirms the modal, the exam is not ongoing or the dates have not changed, the exam is saved.
+     */
+    handleSubmit() {
+        const datesChanged = this.exam.startDate?.diff(this.originalStartDate) !== 0 || this.exam.endDate?.diff(this.originalEndDate) !== 0;
+        if (datesChanged && this.isOngoingExam) {
+            const modalRef = this.modalService.open(ConfirmAutofocusModalComponent, { keyboard: true, size: 'lg' });
+            modalRef.componentInstance.title = 'artemisApp.examManagement.dateChange.title';
+            modalRef.componentInstance.text = this.artemisTranslatePipe.transform('artemisApp.examManagement.dateChange.warning');
+            modalRef.result.then(this.save.bind(this));
+        } else {
+            this.save();
+        }
+    }
+
+    /**
+     * Saves the exam and navigates to the detail page of the exam if the save was successful.
+     * If the save was not successful, an error is shown to the user.
+     */
     save() {
         this.isSaving = true;
 
-        if (this.isImport) {
+        this.upsertOrImportExam()
+            ?.pipe(map((response: HttpResponse<Exam>) => response.body!))
+            .subscribe({
+                next: this.onSaveSuccess.bind(this),
+                error: this.onSaveError.bind(this),
+            });
+    }
+
+    /**
+     * Creates, updates or imports the exam depending on the current state of the component.
+     * @private
+     */
+    private upsertOrImportExam() {
```

Comment:
`upsert` sounds like a typo

Reply:
Actually that's a common term for DB `insert or update if entry exists` (i.e. https://wiki.postgresql.org/wiki/UPSERT), . Can change it however to `createOrUpdateOrImportExam` or sth. like this. Used upsert over the past years so it's natural for me but I understand that it does not fit into this codebase

Reply:
renamed to `createOrUpdateOrImportExam`